addons/stock/models/stock_scrap.py

```diff
@@ -25,6 +25,7 @@ def _get_default_location_id(self):
         'Reference',  default=lambda self: _('New'),
         copy=False, readonly=True, required=True,
         states={'done': [('readonly', True)]})
+    company_id = fields.Many2one('res.company', string='Company', default=lambda self: self.env.user.company_id)
```

Comment:
maybe should be added `related='picking_id.company_id', readonly=True` 🤔

Reply:
In v12+, related are readonly by default.